src/vos/vos_aggregate.c

```diff
@@ -112,7 +112,7 @@ struct agg_io_context {
 	unsigned int		 ic_seg_max;
 	unsigned int		 ic_seg_cnt;
 	/* Reserved SCM extents for new physical entries */
-	struct vos_rsrvd_scm	 ic_rsrvd_scm;
+	struct vos_rsrvd_scm	*ic_rsrvd_scm;
```

Comment:
If you keep vos_rsrvd_scm and vos_scm_publish() unchanged, no changes are required to vos_aggregate.c.

Please keep vos_publish_scm/blocks() as basic scm/nvme publish/cancel functions (no dtx involved), and build dtx logic in their callers.

Reply:
Taking over reservation information from update/aggregation is the most efficient and convenient way, otherwise, we need to allocate DRAM and copy related reservation information when call publish when vos_update_end(), and then specially handle the reservation information when vos_ioc_destroy(). That is more possible to cause issue since vos_update_end() may call update_cancel(), then vos_ioc_destroy() may face different situations.